### PR TITLE
Decouple mutation of App Gateway and AKS settings

### PR DESCRIPTION
**Pull request by draychev**
This PR proposes a change to the `EventProcessor` interface in order to step by step decouple the functionality of AGIC, which mutates Kubernetes Ingress resources from that which mutates App Gateway config.

The goal would be to call 2 separate event processors within `worker.go`:

 -  `w.MutateAppGateway(lastEvent)`
 -  `w.MutateAKS(event)`

Also in this PR: moving `getPublicIPAddress`, `updateIPAddressMap`, `updateIngressStatus` into `pkg/controller/mutate_aks.go`

This change is a follow-up of the conversation here: https://github.com/Azure/application-gateway-kubernetes-ingress/pull/335#pullrequestreview-271032263

There are no functional software changes in the current PR. This is purely renaming and setting up the stage to further decouple `updateIngressStatus()` from the rest.

### CODE REVIEW

File: pkg/worker/worker.go [hunk 1]
```diff
@@ -34,7 +34,11 @@ func (w *Worker) Run(work chan events.Event, stopChannel chan struct{}) {
 	for {
 		select {
 		case event := <-work:
-			if shouldProcess, reason := w.ShouldProcess(event); !shouldProcess {
+			if err := w.MutateAKS(event); err != nil {
```
Comment (akshaysngupta) [inline pkg/worker/worker.go]: `MutateAKS` can be called after `ShouldProcess`. ShouldProcess skips events which are of no interest to the ingress controller as whole.

File: pkg/worker/types.go [hunk 2]
```diff
@@ -11,11 +11,12 @@ import (
...
 type EventProcessor interface {
-	Process(events.Event) error
-	ShouldProcess(events.Event) (bool, *string)
+	MutateAppGateway(events.Event) error
+	MutateAKS(events.Event) error
+	ShouldMutateAppGateway(events.Event) (bool, *string)
 }
 
-// Worker listens on the eventChannel and runs the EventProcessor.Process
+// Worker listens on the eventChannel and runs the EventProcessor.MutateAppGateway
```
Comment (akshaysngupta) [inline pkg/worker/types.go]: This comments should be `Worker listens on the eventChannel and runs the EventProcessor.MutateAppGateway and EventProcessor.MutateAKS

File: pkg/worker/worker.go (hunk 1 above)
Comment (draychev) [reply pkg/worker/worker.go]: `ShouldProcess` is now renamed to `ShouldMutateAppGateway`.
I'm preserving the functionality of `ShouldMutateAppGateway`, which I find to be strictly related to App Gateway stuff.

`MutateAKS` will make its own determinations on whether to mutate AKS or not.

Comment (draychev) [reply pkg/worker/worker.go]: Actually - let me revert that back.  We can simplify this at a later stage - not to boil the ocean in one PR.

**Comment by Morriz** [comment]
when will the IP fix with `c.updateIPAddressMap` make it into a release? That fix is from august. Currently no IP will become visible in the k8s ingresses

**Comment by akshaysngupta** [comment]
@Morriz What version are you running currently. Can you check AGIC logs to see if AGIC has any auths issues with reading public IP address resource associated with Application Gateway ?

**Comment by Morriz** [comment]
Nevermind, I think it was something else that blocked the intended flow.

**Pull request merged**